### PR TITLE
ci: Add Conventional Commits

### PR DESCRIPTION
**Pull request by morning4coffe-dev**
GitHub Issue (If applicable): #659

## PR Type

What kind of change does this PR introduce?

- Build or CI related changes

## What is the new behavior?

A Conventional Commits check was added according to our other  repositories.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested on iOS.
- [ ] Tested on Wasm.
- [ ] Tested on Android.
- [ ] Tested on UWP.
- [ ] Tested in both **Light** and **Dark** themes.
- [x] Associated with an issue (GitHub or internal)

### CODE REVIEW

**Review by Xiaoy312 — APPROVED** [review]

File: .github/workflows/conventional-commits.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name: Conventional Commits
+
+on:
+  pull_request:
+    branches:
+      - master
```
Comment (Youssef1313) [inline .github/workflows/conventional-commits.yml:6]: Do we want to run it for all branches?

File: .github/workflows/conventional-commits.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (Youssef1313) [inline .github/workflows/conventional-commits.yml:9]: Is it needed?

File: .github/workflows/conventional-commits.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+jobs:
+  commitsar:
+    name: Check all commits
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v1
```
Comment (Youssef1313) [inline .github/workflows/conventional-commits.yml:17]:
```suggestion
      - uses: actions/checkout@v3
```

File: .github/workflows/conventional-commits.yml (hunk 2 above)
Comment (morning4coffe-dev) [reply .github/workflows/conventional-commits.yml:9]: @Youssef1313 This is only necessary for private repos. But we use it on our other repos like this.

File: .github/workflows/conventional-commits.yml (hunk 1 above)
Comment (morning4coffe-dev) [reply .github/workflows/conventional-commits.yml:6]: @Youssef1313 Why would we not run it on all?

File: .github/workflows/conventional-commits.yml (hunk 2 above)
Comment (Youssef1313) [reply .github/workflows/conventional-commits.yml:9]: I think we can remove it then if not necessary for public repos

File: .github/workflows/conventional-commits.yml (hunk 1 above)
Comment (Youssef1313) [reply .github/workflows/conventional-commits.yml:6]: @morning4coffe-dev Exactly my question :)

Currently this only run against master, but not in feature branches

Comment (morning4coffe-dev) [reply .github/workflows/conventional-commits.yml:6]: @morning4coffe-dev My bad :) I wanted to ask why aren't we running it in our other repos on all branches?

Comment (Youssef1313) [reply .github/workflows/conventional-commits.yml:6]: cc @jeromelaban

Comment (jeromelaban) [reply .github/workflows/conventional-commits.yml:6]: It's probably best to have it running as a job in azure devops, like this one: https://github.com/unoplatform/uno/blob/master/build/ci/.azure-devops-commitsar.yml

**Review by MartinZikmund — APPROVED** [review]

**Pull request merged**